source/position/position.go

```diff
@@ -0,0 +1,82 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package position
+
+import (
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/jackc/pglogrepl"
+)
+
+//go:generate stringer -type=Type -trimprefix Type
+
+type Type int
+
+const (
+	TypeInitial Type = iota
+	TypeSnapshot
+	TypeCDC
+)
+
+type Position struct {
+	Type     Type              `json:"type"`
+	Snapshot SnapshotPositions `json:"snapshot,omitempty"`
```

Comment:
yea, I went back and forth between these, since a snapshot can contain N tables, I decided to pluralize the positions part, as in multiple positions before multiple tables. what do you think?

Reply:
@lyuboxa Oh, I meant the variable name (not the type). I'm guessing that maybe `Snapshot` is still more appropriate, right?

Reply:
I renamed it here https://github.com/ConduitIO/conduit-connector-postgres/pull/132/commits/ebdec835649ab533e1a52115b21c7c027754a20d.